presto-main/src/main/java/com/facebook/presto/server/BasicQueryInfo.java

```diff
@@ -145,6 +146,23 @@ public BasicQueryInfo(QueryInfo queryInfo)
                 queryInfo.getWarnings());
     }
 
+    public static BasicQueryInfo dispatchedQueryInfo(QueryId queryId, String query, Session session)
+    {
+        return new BasicQueryInfo(
+                queryId,
+                session.toSessionRepresentation(),
+                Optional.empty(),
+                QUEUED,
+                GENERAL_POOL,
+                false,
+                URI.create(""),
+                query,
+                immediateFailureQueryStats(),
```

Comment:
it's better to create a parallel method for queuedQueryStats() and return that instead of reusing failure query stats here. This might cause confusion in future or may lead to issue given the method name suggests it's been used for immediately failed query. 
Or rename the method to emptyQueryStatus or noQueryStats.